tensorboard/webapp/metrics/store/metrics_store_internal_utils.ts

```diff
@@ -153,3 +166,125 @@ export function getRunIds(
   const tagToRunIds = tagMetadata[plugin].tagToRuns;
   return tagToRunIds.hasOwnProperty(tag) ? tagToRunIds[tag] : [];
 }
+
+/**
+ * Returns whether the CardMetadata exactly matches the pinned card from
+ * storage.
+ */
+function cardMatchesCardUniqueInfo(
+  cardMetadata: CardMetadata,
+  cardUniqueInfo: CardUniqueInfo
+): boolean {
+  const noRunId = !cardMetadata.runId && !cardUniqueInfo.runId;
+  return (
+    cardMetadata.plugin === cardUniqueInfo.plugin &&
+    cardMetadata.tag === cardUniqueInfo.tag &&
+    cardMetadata.sample === cardUniqueInfo.sample &&
+    (cardMetadata.runId === cardUniqueInfo.runId || noRunId)
+  );
+}
+
+/**
+ * Attempts to resolve the imported pins against the list of non-pinned cards
```

Comment:
Not a big comment person but let's codified in the comment that the contract is not to sanitize/clip/validate unresolved cards.

Reply:
Done.